### PR TITLE
[WIP] Documenting java files in OFF

### PR DESCRIPTION
**Pull request by vanshikaarora**
## Description

Added documentation for:

- AdditiveFragmentHelper.java
- FileDownloader.java
-  CategoryFragmentViewModel.java
- WikidataApiClient.java

## Related issues and discussion
Document java files which require documentation #2715

### CODE REVIEW

**Comment by vanshikaarora** [comment]
@teolemon @deniger @VaiTon Can you please share some feedback's for this PR, so that I can improve more in further commits :)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java [hunk 1]
```diff
@@ -30,6 +30,14 @@ private AdditiveFragmentHelper(){
...
     }
 
+    /**
+     * Set additive tag on the TextView additiveProduct
+     * @param additives list of Additive names
+     * @param additiveProduct TextView where additive tags should be displayed
+     * @param apiClientForWikiData object of class WikidataApiClient
+     * @param fragment contains a reference to the calling fragment (BaseFragment)
+     * */
+
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java]: Please empty line before method

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java [hunk 2]
```diff
@@ -46,6 +54,13 @@ public static void showAdditives(List<AdditiveName> additives, TextView additive
...
     }
 
+    /**
+     * Returns Additive Tag from the Additive name using the WikidataApi
+     * @param additive the Additive name
+     * @param apiClientForWikiData object of class WikidataApiClient
+     * @param fragment contains a reference to the calling fragment (BaseFragment)
+     * */
+
     private static CharSequence getAdditiveTag(AdditiveName additive, final WikidataApiClient apiClientForWikiData, BaseFragment fragment) {
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java]: Please ensure that you also add the `@return` for methods

**Comment by teolemon** [comment]
@vanshikaarora can you rebase ? If no update, I propose closing

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java [hunk 3]
```diff
@@ -30,6 +30,14 @@ private AdditiveFragmentHelper(){
         //helper class
     }
 
+    /**
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java:33]:
```suggestion
    /**
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java (hunk 1 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java:40]:
```suggestion
     * */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/WikidataApiClient.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
 import retrofit2.converter.jackson.JacksonConverterFactory;
 
 /**
- * Created by Shubham Vishwakarma on 14.03.18.
+ * API client to recieve data from WikiData APIs
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/WikidataApiClient.java:24]:
```suggestion
 * API client to receive data from WikiData APIs
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java [hunk 5]
```diff
@@ -29,6 +29,13 @@
     private AdditiveFragmentHelper(){
         //helper class
     }
+    /**
+     * Set additive tag on the TextView additiveProduct
+     * @param additives list of Additive names
+     * @param additiveProduct TextView where additive tags should be displayed
+     * @param apiClientForWikiData object of class WikidataApiClient
+     * @param fragment contains a reference to the calling fragment (BaseFragment)
+     * */
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AdditiveFragmentHelper.java:39]:
```suggestion
     * */
```

**Pull request merged**